### PR TITLE
add graphql field to query location status

### PR DESCRIPTION
**Pull request by prha**
### Summary & Motivation
CodeLocationStatusQuery is one of our most frequently called queries.  This PR creates a new top-level GraphQL field to retrieve the location status without loading the full repository locations.

### How I Tested These Changes
GraphQL test

### CODE REVIEW

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10815** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10815" target="_blank">Graphite</a>  👈
    * **PR #10820** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10820" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10815?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_workspace.py [hunk 1]
```diff
@@ -125,3 +144,45 @@ def test_load_workspace(self, graphql_context):
                     or "module_name" in metadata_dict
                     or "package_name" in metadata_dict
                 )
+
+    def test_load_location_statuses(self, graphql_context):
+        # Add an error origin
+        original_origins = location_origins_from_yaml_paths(
+            [file_relative_path(__file__, "multi_location.yaml")]
+        )
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_workspace.py:152]: is this an error origin or is this a dangling todo?

File: python_modules/dagster-graphql/dagster_graphql/schema/external.py [hunk 2]
```diff
@@ -107,6 +107,37 @@ class Meta:
...
+    loadStatus = graphene.NonNull(GrapheneRepositoryLocationLoadStatus)
+
+    class Meta:
+        name = "WorkspaceLocationStatusEntry"
+
+    def __init__(self, name, load_status):
+        super().__init__(name=name, loadStatus=load_status)
+
+    def resolve_id(self, _):
+        return self.name
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql/schema/external.py:122]: is this what the other workspace type uses?

Comment (prha) [reply python_modules/dagster-graphql/dagster_graphql/schema/external.py:122]: yeah

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**